cpp/src/table/row_operators.cu

```diff
@@ -133,10 +140,13 @@ table_view remove_struct_child_offsets(table_view table)
  *  |   |
  *  i   f
  *
- * When list columns are present, the decomposition is performed similarly to pure structs but list
- * parent columns are NOT pruned
+ * In the case of structs column with a lists column as its first child such as
+ * `Struct<List<int>, float>`, after decomposition we get three columns `Struct<>`,
+ * `List<int>`, and `float`.
  *
- * For example, if the original table has a column `List<Struct<int, float>>`,
+ * When list columns are present, depending on the input flag `decompose_lists`, the decomposition
```

Comment:
Can you also explain how list columns are decomposed when it's `true` or `false`? `true` is used for equality comparison as you mention below, but what does the decomposition process entail in that case?